dotnet/src/Agents/Abstractions/AgentChat.cs

```diff
@@ -34,26 +50,51 @@ public abstract class AgentChat
     /// <param name="agent">An optional agent, if requesting an agent history.</param>
     /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
     /// <returns>The message history</returns>
-    public IAsyncEnumerable<ChatMessageContent> GetChatMessagesAsync(Agent? agent = null, CancellationToken cancellationToken = default)
+    public async IAsyncEnumerable<ChatMessageContent> GetChatMessagesAsync(
+        Agent? agent = null,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
     {
-        if (agent == null)
+        this.ThrowIfActive();
+
+        try
         {
-            return this.History.ToDescendingAsync();
-        }
+            IAsyncEnumerable<ChatMessageContent>? messages = null;
+
+            if (agent == null)
+            {
+                // Provide primary history
+                messages = this.History.ToDescendingAsync();
+            }
+            else // else provide channel specific history
+            {
+                // Retrieve the requested channel, if exists, and block until channel is synchronized.
+                string channelKey = this.GetAgentHash(agent);
+                AgentChannel? channel = await this.SynchronizeChannelAsync(channelKey, cancellationToken).ConfigureAwait(false);
+                if (channel != null)
+                {
+                    messages = channel.GetHistoryAsync(cancellationToken);
+                }
+            }
 
-        var channelKey = this.GetAgentHash(agent);
-        if (!this._agentChannels.TryGetValue(channelKey, out var channel))
+            if (messages != null)
+            {
+                await foreach (ChatMessageContent message in messages)
+                {
+                    yield return message;
+                }
+            }
+        }
+        finally
         {
-            return Array.Empty<ChatMessageContent>().ToAsyncEnumerable();
+            this.ClearActivitySignal();
         }
-
-        return channel.GetHistoryAsync(cancellationToken);
     }
 
     /// <summary>
-    /// Append messages to the conversation.
+    /// Append a message to the conversation.  Adding a message while an agent
+    /// is active is not allowed.
     /// </summary>
-    /// <param name="message">Set of non-system messages with which to seed the conversation.</param>
+    /// <param name="message">A of non-system messages with which to append to the conversation.</param>
```

Comment:
Please update the comment, looks like a bad edit

Reply:
Was supposed to be the _singular_ (vs _plural_) version...fixed.